processor/src/main/java/org/mapstruct/ap/internal/model/source/PropertyEntry.java

```diff
@@ -115,4 +107,39 @@ public String getFullName() {
         return Strings.join( Arrays.asList(  fullName ), "." );
     }
 
+    public PropertyEntry pop() {
+        if ( fullName.length > 1 ) {
+            String[] newFullName = Arrays.copyOfRange( fullName, 1, fullName.length );
```

Comment:
Why didn't we make `fullName` a `List<String>`? We could have used `List.subList(...)` here instead then, avoiding the array-copy,